test/metabase/notification/send_test.clj

```diff
@@ -184,3 +187,61 @@
               (is (prometheus-test/approx= 1 (mt/metric-value system :metabase-notification/channel-send-error
                                                               {:payload-type "notification/testing"
                                                                :channel-type "channel/metabase-test"}))))))))))
+
+(deftest notification-dispatcher-test
```

Comment:
I think we want some parallel stress tests on this for peace of mind. 

You could take some inspiration from our existing queues https://github.com/metabase/metabase/blob/3d6c90698b1f2ece39bd5cc000c7eba757653d36/test/metabase/util/queue_test.clj